e2e/config/azure.go

```diff
@@ -600,3 +608,74 @@ func (a *AzureClient) createVMSS(ctx context.Context, resourceGroupName string,
 	return &vmssResp.VirtualMachineScaleSet, nil
 
 }
+
+// GetLatestVMExtensionImageVersion lists VM extension images for a given extension name and returns the latest version.
+// This is equivalent to: az vm extension image list -n Compute.AKS.Linux.AKSNode --latest
+func (a *AzureClient) GetLatestVMExtensionImageVersion(ctx context.Context, location, extType, extPublisher string) (string, error) {
+	// List extension versions
+	resp, err := a.VMExtensionImages.ListVersions(ctx, location, extPublisher, extType, &armcompute.VirtualMachineExtensionImagesClientListVersionsOptions{})
+	if err != nil {
+		return "", fmt.Errorf("listing extension versions: %w", err)
+	}
+
+	if len(resp.VirtualMachineExtensionImageArray) == 0 {
+		return "", fmt.Errorf("no extension versions found")
+	}
+
+	version := make([]Version, len(resp.VirtualMachineExtensionImageArray))
+	for i, ext := range resp.VirtualMachineExtensionImageArray {
+		version[i] = parseVersion(ext)
+	}
+
+	sort.Slice(version, func(i, j int) bool {
+		return version[i].Less(version[j])
+	})
+
+	return *version[len(version)-1].Original.Name, nil
+}
+
+// Version represents a parsed version of a VM extension image.
+type Version struct {
```

Comment:
nit: VMExtenstionVersion?

Reply:
Done, PTAL.

Reply:
wait didnt I provided the same review ? :D where is my comment !

Reply:
It was here: https://github.com/Azure/AgentBaker/pull/6533#discussion_r2169456271

I extracted the code from there, into this non-GPU checks: https://github.com/Azure/AgentBaker/pull/6533#discussion_r2169456271